src/datasets.jl

```diff
@@ -679,7 +680,7 @@ function _open_dataset_version(name; auth::Authentication=__auth__())::_RESTResp
     _restcall(auth, :POST, "user", "datasets", name, "versions")
 end
 
-function _upload_dataset(upload_config, local_path)
+function _upload_dataset(upload_config, local_path; progress=true)
```

Comment:
I feel it's better if we don't duplicate the default value though. This function is only called in one spot right now, so no other call sites would need to be modified:

```suggestion
function _upload_dataset(upload_config, local_path; progress::Bool)
```